ykcs11/mechanisms.c

```diff
@@ -706,6 +706,17 @@ CK_RV check_pvtkey_template(gen_info_t *gen, CK_MECHANISM_PTR mechanism, CK_ATTR
       break;
 
     case CKA_SENSITIVE:
+      if (*((CK_BBOOL *)templ[i].pValue) != CK_TRUE) {
+        DBG("CKA_SENSITIVE must be TRUE or omitted");
+        return CKR_ATTRIBUTE_VALUE_INVALID;
+      }
+
```

Comment:
There is still one issue with this code: case fall-through. If sensitive is set to true this code goes ahead to ` case CKA_EXTRACTABLE` and there it fails. (because first check checks for != TRUE and the second for != FALSE this means that setting sensitive to any value will fail with either `CKA_SENSITIVE must be TRUE or omitted` or `CKA_EXTRACTABLE must be FALSE or omitted`)

Reply:
Third time the charm?

Reply:
Yep, this time it really works! Approved :+1: Thanks for your time @a-dma !

Reply:
Thank you for your help!